### PR TITLE
Add topography option to support Hughes2023 double-mountain config

### PR DESCRIPTION
**Pull request by akshaysridhar**

### CODE REVIEW

**Comment by szy21** [comment]
What's the status of this PR?

File: config/model_configs/sphere_baroclinic_wave_rhoe_hughes2023.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+dt_save_state_to_disk: "2days"
+initial_condition: "DryHughes2023BaroclinicWave"
+topography: "Hughes2023"
+topo_smoothing: false
+implicit_diffusion: true
+approximate_linear_solve_iters: 2
+moist: "equil"
+dt: "200secs"
+t_end: "10days"
+ode_algo: SSPKnoth
```
Comment (szy21) [inline config/model_configs/sphere_baroclinic_wave_rhoe_hughes2023.yml:10]:
```suggestion
```

File: config/model_configs/sphere_baroclinic_wave_rhoe_hughes2023.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+vert_diff: "FriersonDiffusion"
```
Comment (szy21) [inline config/model_configs/sphere_baroclinic_wave_rhoe_hughes2023.yml:11]:
```suggestion
```
I don't think baroclinic wave has vertical diffusion?

File: config/model_configs/sphere_baroclinic_wave_rhoe_hughes2023.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+dt_save_state_to_disk: "2days"
+initial_condition: "DryHughes2023BaroclinicWave"
+topography: "Hughes2023"
+topo_smoothing: false
+implicit_diffusion: true
+approximate_linear_solve_iters: 2
```
Comment (szy21) [inline config/model_configs/sphere_baroclinic_wave_rhoe_hughes2023.yml:6]:
```suggestion
```

File: config/model_configs/sphere_baroclinic_wave_rhoe_hughes2023.yml (hunk 1 above)
Comment (szy21) [reply config/model_configs/sphere_baroclinic_wave_rhoe_hughes2023.yml:10]: (This is the default)

File: src/initial_conditions/initial_conditions.jl [hunk 4]
```diff
@@ -421,6 +421,109 @@ function shallow_atmos_baroclinic_wave_values(z, ϕ, λ, params, perturb)
     return (; T_v, p, u, v)
 end
 
+function hughes2023_baroclinic_wave_values(
+    z,
+    ϕ,
+    λ,
+    params,
+    perturb,
+    deep_atmosphere,
```
Comment (szy21) [inline src/initial_conditions/initial_conditions.jl:430]: Where is deep_atmosphere used in this function?

File: src/initial_conditions/initial_conditions.jl [hunk 5]
```diff
@@ -421,6 +421,109 @@ function shallow_atmos_baroclinic_wave_values(z, ϕ, λ, params, perturb)
...
+            (-sind(ϕ_c) * cosd(ϕ) + cosd(ϕ_c) * sind(ϕ) * cosd(λ - λ_c)) /
+            sin(r / R) * cond
+        v +=
+            16 * V_p / 3 / sqrt(FT(3)) *
+            F_z *
+            c3 *
+            s1 *
+            cosd(ϕ_c) *
+            sind(λ - λ_c) / sin(r / R) * cond
+    end
```
Comment (szy21) [inline src/initial_conditions/initial_conditions.jl:500]: Is this the same as other baroclinic wave cases? Could we use the other function?

File: .buildkite/pipeline.yml [hunk 6]
```diff
@@ -380,6 +380,13 @@ steps:
           --config_file $CONFIG_PATH/sphere_baroclinic_wave_rhoe_topography_dcmip_rs.yml
           --job_id sphere_baroclinic_wave_rhoe_topography_dcmip_rs
         artifact_paths: "sphere_baroclinic_wave_rhoe_topography_dcmip_rs/output_active/*"
+      
+      - label: ":computer: baroclinic wave (ρe) topography (hughes2023)"
```
Comment (szy21) [inline .buildkite/pipeline.yml:384]: Could we remove the dcmip case?

**Review by szy21 — COMMENTED** [review]
Thanks! I didn't read the detailed initial condition and topography. Here are some comments on the setup and code structure.

File: config/model_configs/sphere_baroclinic_wave_rhoe_hughes2023.yml (hunk 2 above)
Comment (akshaysridhar) [reply config/model_configs/sphere_baroclinic_wave_rhoe_hughes2023.yml]: Fixed.

File: src/initial_conditions/initial_conditions.jl (hunk 5 above)
Comment (akshaysridhar) [reply src/initial_conditions/initial_conditions.jl]: Bad rebase, fixed.

File: config/model_configs/sphere_baroclinic_wave_rhoe_hughes2023.yml (hunk 1 above)
Comment (akshaysridhar) [reply config/model_configs/sphere_baroclinic_wave_rhoe_hughes2023.yml:8]: Ah yes, I've updated the help string in the default config.

File: config/model_configs/sphere_baroclinic_wave_rhoe_hughes2023.yml [hunk 7]
```diff
@@ -0,0 +1,16 @@
+dt_save_state_to_disk: "2days"
+initial_condition: "DryBaroclinicWave"
```
Comment (szy21) [inline config/model_configs/sphere_baroclinic_wave_rhoe_hughes2023.yml:2]: Should this be MoistBaroclinicWave? Also are the initial condition exactly the same, or is there any perturbation related to topography?

File: config/model_configs/sphere_baroclinic_wave_rhoe_hughes2023.yml
Comment (szy21) [inline config/model_configs/sphere_baroclinic_wave_rhoe_hughes2023.yml:1]: Could you add this experiment to either ci or longrun?

File: examples/Manifest-v1.11.toml
Comment (szy21) [inline examples/Manifest-v1.11.toml:1]: Should we revert this change?

**Review by szy21 — APPROVED** [review]

File: examples/Manifest-v1.11.toml
Comment (akshaysridhar) [reply examples/Manifest-v1.11.toml:1]: Yes - the Manifest will follow the 1.10.5 version ( I locally have 1.11 installed as well so I added this for tests)
